receiver/googlecloudpubsubreceiver/receiver.go

```diff
@@ -111,6 +111,9 @@ func (receiver *pubsubReceiver) Start(ctx context.Context, _ component.Host) err
 }
 
 func (receiver *pubsubReceiver) Shutdown(_ context.Context) error {
+	if receiver.handler == nil {
+		return nil
+	}
 	receiver.logger.Info("Stopping Google Pubsub receiver")
 	receiver.handler.CancelNow()
```

Comment:
the check for nil might be better around this call to still have the logs?

Reply:
I don’t think we are stopping anything if we didn’t start, so no logging might be best. Codeowners might know best.

Reply:
Indeed, the handler is the thing that needs to be stopped and will consume resources. So nothing needs to be logged.